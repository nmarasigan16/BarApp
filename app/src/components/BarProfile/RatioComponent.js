import React, { Component } from 'react';
import PropTypes from 'prop-types';
import {
    View,
    Dimensions,
} from 'react-native';
import Icon from 'react-native-vector-icons/FontAwesome';
import Text from '../general/text/TextComponent';
import Number from '../general/number/NumberComponent';
import { colorScheme } from "../../lib/styles/ColorScheme";
import { responsiveFontSize } from 'react-native-responsive-dimensions';


const RatioComponent = ({guys}) => {

    return (
        <View style={{
            flexDirection: 'column',
            alignItems: 'center',
            justifyContent: 'center',
            flex: 1
        }}>
            <Text
                style={{
                    marginBottom: 10,
                    fontSize: responsiveFontSize(2),
                    color: colorScheme.accent,
                }}>
                Ratio
            </Text>
            <View style={{
                flexDirection: 'row',
                alignItems: 'center',
                justifyContent: 'center'
            }}>
                <View style={{
                    flex: 1,
                    flexDirection: 'column',
                    justifyContent: 'center',
                    alignItems: 'center',
                }}>
                    <Text
                        style={{
                            color: 'deepskyblue',
                            fontSize: responsiveFontSize(3)
                        }}>
                        {guys}%
                    </Text>
                    <Icon
                        name={'mars'}
                        color={'deepskyblue'}
                        size={responsiveFontSize(2)}
                    />
                </View>
                <View style={{
                    flex: 1,
                    flexDirection: 'column',
                    justifyContent: 'center',
                    alignItems: 'center',
                }}>
                    <Text
                        style={{
                            color: 'deeppink',
                            fontSize: responsiveFontSize(3)
                        }}>
                        {100-guys}%
                    </Text>
                    <Icon
                        name={'venus'}
                        color={'deeppink'}
                        size={responsiveFontSize(2)}
                    />
                </View>
            </View>
        </View>
    )
};

export default RatioComponent;
