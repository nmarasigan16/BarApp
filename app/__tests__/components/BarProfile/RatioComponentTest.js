import 'react-native';
import React from 'react';
import RatioComponent from '../../../src/components/BarProfile/RatioComponent';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

it('renders correctly', () => {
    const tree = renderer.create(
        <RatioComponent guys={10}/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
});
