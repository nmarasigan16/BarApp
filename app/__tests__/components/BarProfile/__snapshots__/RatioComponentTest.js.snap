// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "fontFamily": "Courier New",
        },
        Object {
          "color": "silver",
          "fontSize": 30.607554623001164,
          "marginBottom": 10,
        },
      ]
    }
  >
    Ratio
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "deepskyblue",
              "fontSize": 45.911331934501746,
            },
          ]
        }
      >
        10
        %
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "deepskyblue",
              "fontSize": 30.607554623001164,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "column",
          "justifyContent": "center",
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "fontFamily": "Courier New",
            },
            Object {
              "color": "deeppink",
              "fontSize": 45.911331934501746,
            },
          ]
        }
      >
        90
        %
      </Text>
      <Text
        accessible={true}
        allowFontScaling={false}
        disabled={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "deeppink",
              "fontSize": 30.607554623001164,
            },
            undefined,
            Object {
              "fontFamily": "FontAwesome",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
